src/core/mount.c

```diff
@@ -160,6 +160,8 @@ static void mount_init(Unit *u) {
          * already trying to comply its last one. */
         m->exec_context.same_pgrp = true;
 
+        m->exec_context.mount_apivfs = true;
```

Comment:
hmm, actually, this doesn't look right... right now if the bool is set this has the effect that a mount namespace is created... (see exec_needs_mount_namespace()) which we actually don#t want here. (in particular not for mount units...) my bad, sorry for the confusion

I figure we have to have this as two bools after all, my bad. When the bool is unset turn it on but do not enable the namespace logic, when the bool is true enable the namespace logic and turn it off, and when is off do not propagate that to the namespace logic.